### PR TITLE
[MAINT] fix various typos

### PR DESCRIPTION
**Pull request by luzpaz**
Changes proposed in this pull request:

- Fixed some user-facing and non-user-facing typos

### CODE REVIEW

**Review by NicolasGensollen — COMMENTED** [review]
LGTM so far, thanks @luzpaz !
Please remove the [WIP] tag when the PR is ready.

**Review by bthirion — APPROVED** [review]
Many thx !

**Review by NicolasGensollen — APPROVED** [review]
Thanks !

**Review by ymzayek — COMMENTED** [review]
LGTM! As a first-time contributor, can you add your name to `doc/changes/names.rst`?

**Comment by luzpaz** [comment]
Thanks, done in [d40a0c2](https://github.com/nilearn/nilearn/pull/3350/commits/d40a0c24be418fdab658ad14119b50131df3d8d6)

File: doc/changes/names.rst [hunk 1]
```diff
@@ -134,6 +134,8 @@
 
 .. _Loic Tetrel: https://ltetrel.github.io/
 
+.._Luz Paz: https://github.com/luzpaz
```
Comment (ymzayek) [inline doc/changes/names.rst:137]: Thx, just one small thing:

```suggestion
.. _Luz Paz: https://github.com/luzpaz
```

Comment (luzpaz) [reply doc/changes/names.rst]: :blush:

**Comment by luzpaz** [comment]
Shall I squash or defer to the merge dev?

**Comment by ymzayek** [comment]
@luzpaz all looks good but we'll wait until all the tests run and pass. Then I will do a final check and merge it!

**Comment by ymzayek** [comment]
Ok merging

**Pull request merged**